datalad/distribution/get.py

```diff
@@ -49,53 +51,209 @@
+        else:
+            # for everything else we are interested in the container
+            d = dirname(path)
+            if not d:
+                d = curdir
+        # this could be `None` if there is no git repo
+        dspath = dir_lookup.get(d, GitRepo.get_toppath(d))
+        dir_lookup[d] = dspath
+        if not dspath:
+            lgr.warning("%s is not part of a dataset, ignored.", path)
+            continue
+        if isdir(path):
+            ds = Dataset(dspath)
+            # we need to doublecheck that this is not a subdataset mount
+            # point, in whic case get_toppath() would point to the parent
+            smpath = ds.get_containing_subdataset(
+                path, recursion_limit=1).path
+            if smpath != dspath:
+                # fix entry
+                dir_lookup[d] = smpath
+                # submodule still needs to be obtained
+                unavailable_paths.append(path)
+                continue
+            if recursive:
+                # make sure we get everything relevant in all _checked out_
+                # subdatasets, obtaining of previously unavailable subdataset
+                # else done elsewhere
+                subs = ds.get_subdatasets(fulfilled=True,
+                                          recursive=recursive,
+                                          recursion_limit=recursion_limit)
+                for sub in subs:
+                    subdspath = opj(dspath, sub)
+                    if subdspath.startswith(_with_sep(path)):
+                        # this subdatasets is underneath the search path
+                        # we want it all
+                        out[subdspath] = [subdspath]
+        out[dspath] = out.get(dspath, []) + [path]
+    return out, unavailable_paths, dir_lookup
+
+
+def _get(content_by_ds, refpath=None, source=None, jobs=None,
+         fulfill_datasets=False):
+    for ds_path in sorted(content_by_ds.keys()):
+        cur_ds = Dataset(ds_path)
+        content = content_by_ds[ds_path]
+        results = []
+        if len(content) >= 1 and content[0] == curdir:
+            # we hit a subdataset that just got installed few lines above, and was
+            # requested specifically, as opposed to some of its content. Unless we
+            # are asked to fulfill all handles that at some point in the process
+            # we consider having fulfilled the dataset handle good enough
+            results.append(cur_ds)
+            if not fulfill_datasets:
+                lgr.debug(
+                    "Will not get any content in subdataset %s without recursion enabled",
+                    cur_ds)
+                yield results
+                continue
+
+        # needs to be an annex:
+        found_an_annex = isinstance(cur_ds.repo, AnnexRepo)
+        if not found_an_annex:
+            lgr.debug("Found no annex at %s. Skipped.", cur_ds)
+            if results:
+                yield results
+            continue
```

Comment:
interesting, so I stopped with pdb right upon `if` (results was just `[]`), with 'n' it went right away to the top of theloop, so I guess "continue" is not really evaluated at run time